sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusMetricsProvider.cs

```diff
@@ -0,0 +1,238 @@
+                    if (peekedMessages.Count > 0 && activeMessage == null)
+                    {
+                        _logger.LogDebug("{_serviceBusEntityType} {_entityPath} contains multiple messages but none are active in the peeked batch.");
+                    }
+                }
+
+                if (_serviceBusEntityType == ServiceBusEntityType.Queue)
+                {
+                    return await GetQueueMetricsAsync(activeMessage).ConfigureAwait(false);
+                }
+                else
+                {
+                    return await GetTopicMetricsAsync(activeMessage).ConfigureAwait(false);
+                }
+            }
+            catch (ServiceBusException ex)
+            when (ex.Reason == ServiceBusFailureReason.MessagingEntityNotFound)
+            {
+                _logger.LogWarning($"ServiceBus {entityName} '{_entityPath}' was not found.");
+            }
+            catch (UnauthorizedAccessException) // When manage claim is not used on Service Bus connection string
+            {
+                if (TimeToLogWarning())
+                {
+                    _logger.LogWarning($"Connection string does not have Manage claim for {entityName} '{_entityPath}'. Failed to get {entityName} description to " +
+                        $"derive {entityName} length metrics. Falling back to using first message enqueued time.");
+                }
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"Error querying for Service Bus {entityName} scale status: {e.Message}");
+            }
+
+            // Path for connection strings with no manage claim
+            return CreateTriggerMetrics(activeMessage, 0, 0, 0, _isListeningOnDeadLetterQueue);
+        }
+
+        private static bool MessageIsActive(ServiceBusReceivedMessage message)
+        {
+            return message.State == ServiceBusMessageState.Active;
+        }
+
+        private async Task<ServiceBusTriggerMetrics> GetQueueMetricsAsync(ServiceBusReceivedMessage message)
+        {
+            QueueRuntimeProperties queueRuntimeProperties;
+            QueueProperties queueProperties;
+            long activeMessageCount = 0, deadLetterCount = 0;
+            int partitionCount = 0;
+
+            queueRuntimeProperties = await _administrationClient.Value.GetQueueRuntimePropertiesAsync(_entityPath).ConfigureAwait(false);
+            activeMessageCount = queueRuntimeProperties.ActiveMessageCount;
+            deadLetterCount = queueRuntimeProperties.DeadLetterMessageCount;
+
+            // If partitioning is turned on, then Service Bus automatically partitions queues into 16 partitions
+            // See more information here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-partitioning#standard
+            queueProperties = await _administrationClient.Value.GetQueueAsync(_entityPath).ConfigureAwait(false);
```

Comment:
@JoshLove-msft :  Do you know of a way where we can remove the hardcoded 16 for this?

Reply:
It doesn't look like this is exposed in the Admin API, but 16 only applies to basic and standard. For premium, it is configurable on namespace creation.